blog/en/blog/2022/09/08/api-monetization-using-stack.md

```diff
@@ -0,0 +1,150 @@
+
+There are many popular open-source projects available like [Apache APISIX](https://apisix.apache.org/) or alternative enterprise SaaS solutions such as [Azure API Management](https://docs.microsoft.com/en-us/azure/api-management/), [API7 Cloud](https://api7.ai/cloud) in a public cloud. You can investigate the pros and cons of each to choose the more suitable one for your needs.
+
+![Apache APISIX API Gateway](https://dev-to-uploads.s3.amazonaws.com/uploads/articles/yybu74a2ys0f2u6low7z.png)
+
+#### Apply rate limit policies
+
+Resources cost money 💰. We can protect an API by adding _a rate limit policy_ with Apache APISIX as it is a basic step toward API Monetization. Apache APISIX allows you to set throttling limits per each API consumer and quotas to your APIs and allows you to control third-party usage of your API by ensuring you are able to monetize your API.
+
+APISIX uses its `limit-count` (_rate limiting_) plugin. A[PI rate limiting plugin](https://apisix.apache.org/docs/apisix/plugins/limit-count/) can prevent an API not only from being overwhelmed or from possible malicious attacks but also it can enforce a limit on the number of data clients can consume. Later you can charge API consumers by the quantity of data used (the number of requests).
+
+With the help of APISIX `rate-limiting` plugin, you can also configure the different rate limits for authenticated and unauthenticated requests. It also defines the limit quota in the [response headers](https://apisix.apache.org/docs/apisix/plugins/limit-count/) to track the maximum number of requests you are permitted to make or the number of requests remaining in the current rate limit window.
+
+Refer to the documentation to understand [Consumer concept](https://apisix.apache.org/docs/apisix/terminology/consumer/) and learn the different ways to set up [rate limiting](https://apisix.apache.org/docs/apisix/plugins/limit-count/) with Apache APISIX.
+
+### A billing provider
+
+Next, for your API monetization stack, you need a 3rd-party recurring billing solution, such as [Stripe](https://stripe.com/en-gb-ee), [Recurly](https://recurly.com/), [Hypercurrent](https://www.hypercurrent.io/), and many more. But again, we do not recommend any particular payment service in this post and leave the choice of which payment provider to use up to you. The billing provider obviously needs to receive usage charges for each customer, issue an invoice, and support multiple billing models, currencies 💵 💴 💶 💷, and payment methods.
+
```

Comment:
I'd add an additional advice. Most of the time, you don't want to couple yourself to the billing provider for resiliency purpose. Imagine the provider is not available: it would be the responsibility of the plugin to retry until it works. It's not a great approach.

The alternative is to send the data to a component under your control inside your network, _e.g._, Kafka, and then develop a dedicated job to read from Kafka and push to the provide with all bells and whistles of failure management.

Reply:
Sure, I can add your advice to the readers just below the section.

Reply:
It should be a bit more detailed - 2-3 paragraphs.

Honestly, there's no way you can directly talk to the billing provider besides a toy project.

Reply:
Yeah, I will a toy project with the detailed instruction on how to achieve it, and this post just gives an idea:)

<img width="542" alt="image" src="https://user-images.githubusercontent.com/14247607/189070140-90bcde3c-6a42-4adb-a9aa-be706f7a81cd.png">

Maybe there will be contributors to help us to achieve that.